UKHomeOffice/nmsw-ui
# NMSW-1074 Only allow CYA page when passengers complete

# Ticket



----

## AC

As an agent
When I submit a declaration
I want to inform Border force that all checked in passengers have now departed

----

## To test

**Scenarios to test**

- create a new declaration
- upload a FAL1
- continue to task list page
- > passenger status should be required
- > you should not be able to continue to CYA page
- type in CYA url (/report-voyage/check-your-answers?report=<decId>)
- > you should be redirect back to task list page

**NO PASSENGERS**

- click on passengers
- select no
- continue to task list page
- > passenger status should be complete
- > you should be able to continue to CYA page

**YES PASSENGERS > NO FAL 6**

- click on passengers
- select yes
- continue
- leave page
- return to that declaration (task list page)
- > passenger status should be required
- > you should not be able to continue to CYA page
- type in CYA url (/report-voyage/check-your-answers?report=<decId>)
- > you should be redirect back to task list page

**YES PASSENGERS > YES FAL 6 > NOT FINAL**

- click on passengers
- select yes
- upload a valid FAL6
- continue
- click save and come back later (takes to task list page)
- > passenger status should be in progress
- > you should not be able to continue to CYA page
- type in CYA url (/report-voyage/check-your-answers?report=<decId>)
- > you should be redirect back to task list page

 
**YES PASSENGERS > YES FAL 6 > YES FINAL > UPT IS 100%**

- click on passengers
- select yes
- upload a valid FAL6 
- continue
- select this is final file
- click continue
- check UPT status response (in network tab) & we are expecting it to be 100% for this scenario
- > passenger status should be complete
- > you should be able to continue to CYA page

----

## Notes
